backends-velox/src/main/scala/org/apache/spark/sql/catalyst/VeloxAggregateFunctionRewriteRule.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.expressions.Literal
+import org.apache.spark.sql.catalyst.expressions.aggregate.{AggregateExpression, HLLVeloxAdapter, HyperLogLogPlusPlus}
+import org.apache.spark.sql.catalyst.plans.logical.{Aggregate, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.types._
+import io.glutenproject.backendsapi.BackendsApiManager
+import io.glutenproject.GlutenConfig
+
+case class VeloxAggregateFunctionRewriteRule(spark: SparkSession) extends Rule[LogicalPlan] {
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
+        case a: Aggregate =>
+          a.transformExpressions {
+            case hllExpr @ AggregateExpression(hll: HyperLogLogPlusPlus, _, _, _, _)
+              if GlutenConfig.getConf.enableNativeHyperLogLogAggregateFunction &&
+                GlutenConfig.getConf.enableColumnarHashAgg &&
+                !hasDistinctAggregateFunc(a) && isDataTypeSupported(hll.child.dataType) =>
```

Comment:
Hey @zhli1142015 @zhouyuan @rui-mo , I must miss something here, curious why this function doesn't support with `distinct`, is that `PartialMerge` is not supported?

Reply:
I remember PartialMerge is not well supported when doing this PR. this check is removed already in main branch.

Reply:
I see, thank you for point me out!